### PR TITLE
Add type annotations to `celery/utils/nodenames.py`

### PR DESCRIPTION
**Pull request by lyzlisa**
## Description

Relates to #7394

### CODE REVIEW

File: celery/utils/nodenames.py [hunk 1]
```diff
@@ -46,57 +57,62 @@ def worker_direct(hostname):
...
 
-def default_nodename(hostname):
+def default_nodename(hostname: str) -> str:
     """Return the default nodename for this process."""
     name, host = nodesplit(hostname or '')
     return nodename(name or NODENAME_DEFAULT, host or gethostname())
 
 
-def node_format(s, name, **extra):
+def node_format(s: str, name: str, **extra: Any) -> str:
```
Comment (Nusnus) [inline celery/utils/nodenames.py:87]:
```suggestion
def node_format(s: str, name: str, **extra: dict) -> str:
```

File: celery/utils/nodenames.py [hunk 2]
```diff
@@ -46,57 +57,62 @@ def worker_direct(hostname):
...
+
     index = current_process_index()
     return f'{prefix}{index}' if index else default
 
 
 _fmt_process_index_with_prefix = partial(_fmt_process_index, '-', '')
 
 
-def host_format(s, host=None, name=None, **extra):
+def host_format(s: str, host: str | None = None, name: str | None = None, **extra: Any) -> str:
```
Comment (Nusnus) [inline celery/utils/nodenames.py:103]:
```suggestion
def host_format(s: str, host: str | None = None, name: str | None = None, **extra: dict) -> str:
```

**Review by Nusnus — APPROVED** [review]
Fixed a small issue where `Any` was used instead of `dict` for the kwargs `extra`.
Besides looks great!

Good job and thank you for your contribution!

File: celery/utils/nodenames.py [hunk 3]
```diff
@@ -1,13 +1,19 @@
...
 from functools import partial
+from typing import TYPE_CHECKING
 
 from kombu.entity import Exchange, Queue
 
 from .functional import memoize
 from .text import simple_format
 
+if TYPE_CHECKING:
+    from typing import Any
```
Comment (Nusnus) [inline celery/utils/nodenames.py:15]: This can now be removed

**Review by Nusnus — CHANGES_REQUESTED** [review]
Approved. Pending merge on lint fix @lyzlisa

**Review by auvipy — CHANGES_REQUESTED** [review]
should we follow this route or consider stub approach?

**Review by 50-Course — APPROVED** [review]
Thank you correcting the annotations. This look great. Thank you for making this MR

**Comment by 50-Course** [comment]
@auvipy After much consideration, I think instead of annotating the library code directly, would be great enforcing correctness, than using the stubs approach. Looking at `kombu` there's already an ongoing effort using the typing approach as well.

Open to corrections.

**Comment by auvipy** [comment]
I think we should think about the stub approach to kombu as well

**Comment by 50-Course** [comment]
I have marked the old review as outdated. Looking to draft patch later for `kombu`, hopefully I start next week. Happy weekend to you.

**Comment by 50-Course** [comment]
**Update**: I would pass up the first two days of next week as it collides with my birthday, and would spend time with family. See you later.

**Review by Nusnus — APPROVED** [review]
LGTM

**Pull request merged**